### DISCUSSION TITLE
Passive buzzer

### DISCUSSION THREAD

**Discussion by pauljcheffus** [astlouys/Pico-Green-Clock#21] [Q&A]
Hi Andre

The clock is now working well and I decided to make the first modification so I followed the instructions in the user guide for adding a passive buzzer but no sound comes out with the circuit provided. If I connect the buzzer directly between GND and GPIO pin 19 I can get a sound from the buzzer with a simple test script.

Any ideas why the circuit in the user guide isn’t working?

Buzzer is this one
https://shop.pimoroni.com/products/piezo-transducer

Cheers

Paul

**Comment by astlouys** [comment]
Hi Paul,

I suppose you are referring to the schematic on page 66 of the User Guide (Appendix E) ?

If the buzzer works fine when you plug it directly to pin 19, it means that the software works fine and the buzzer too... However, a passive buzzer being relatively low impedance, it is not recommended to connect it directly to the Pico's GPIO. That's why I added a buffer transistor that has also the advantage of allowing to feed the passive buzzer with 5 volts, making a louder sound.

The few things that I can see are:

1) Check your 5 volts DC with a voltmeter to make sure it is good.
2) Make sure you connect together the ground of the clock with the ground of the 5 volts (but keeping the +5 volts connected only to the passive buzzer as indicated on the diagram).
3) If you hear the buzzer on GPIO 19, I assume that you properly enable the #define PASSIVE_BUZZER in the source code ?

The transistor must be a NPN type. 2N2222 or 2N3904 are fine...

Not sure what else could go wrong...  Keep me informed.

André

**Comment by pauljcheffus** [comment]
Hi Andre

I connected the buzzer exactly per the diagram (page 69) and used a 2n2222
transistor. The five volts and gnd connection both come from the Pico W.
Pin 8 for gnd and pin 39 for five volts.

I definitely have five volts between pins 8 and 39 as I checked with a
multimeter.

The clock firmware has been updated to include the buzzer code but in this
instance I tested the buzzer on a separate Pico W and with the transistor
in circuit (wired to exactly the same pins) there is no sound but when the
buzzer is connected directly between gnd and the gpio pin it works.

Cheers

Paul

**Reply by astlouys** [reply]
Hi Paul,

I double check the potential problems...

There seems to have two different pin-outs for the 2N2222. "Micro Electronics" publishes the same pin-out as I put on the schematic, whereas "ON Semiconductor" publishes a different pin-out, with Emitter and Collector reversed...

You may want to try it and / or check the manufacturer of your transistor and double-check its pin-out. Is yours a TO-92 case ?

**Comment by pauljcheffus** [comment]
Hi Andre

Thank you.

Yes it is a TO92 case and I’ve tried it both ways around and still nothing.
I replaced the buzzer with an LED and that flashes as expected via the
transistor.

Maybe the buzzer I have cannot be controlled using a transistor.

Cheers

Paul

**Reply by astlouys** [reply]
Oh ! Are you sure it is a "passive" buzzer ?  They look very similar to the "active" ones (there are hints in the User Guide on how to verify if it is a passive or an active buzzer).  I never tried it myself, but I think you could give it a try by replacing the buzzer with a small speaker and see if you get something.

If it is an active buzzer, there is already an oscillator integrated in the unit. Not sure it likes the idea of being feed with a 500 Hz or 2 kHz source...  ;-(

If the LED works fine, as you wrote, I think you test everything except the buzzer itself... (of course, you can't "see" the frequency variations on the LED since they are still too fast for the eye, except if you change the code to have very low frequencies - I don't remember having put validation to prevent very low frequencies)...

**Comment by pauljcheffus** [comment]
Hi Andre

That’s a good question when I tried to measure the resistance it is
basically OL on the autoranging multimeter I have.

I just tried playing this tune from here and it appears to work when
attached directly to the gpio pin but is quite quiet.

https://www.tomshardware.com/how-to/buzzer-music-raspberry-pi-pico

I’ll have to see if I can find another type.

Cheers

Paul

**Reply by astlouys** [reply]
Yeah... if it shows OL ("overload"), the static resistance is probably too high and it may indicate an active buzzer.

What happens if you connect it directly to 5 volts ? An active buzzer will sound whereas a passive buzzer will only kind of "click".

**Comment by pauljcheffus** [comment]
Hi Andre

It doesn’t make any noise if I connect it to a 5v supply.

Cheers

Paul

**Reply by astlouys** [reply]
So, it should be a passive buzzer, which is what we want.

I haven't used the passive buzzer for a few months now, since I removed it from my development environment. I may try to find one to connect it and make sure the most recent version of code still works fine. However, you said that if you connect it directly to the GPIO, you can hear it. That leaves me perplex...
